### ISSUE TITLE
Some path configurations impassable for guests due to wide path flags

### ISSUE THREAD

**Issue by deurklink** [OpenRCT2/OpenRCT2#22824]
### Operating System

Windows 10

### OpenRCT2 build

v0.4.14 (but has been in game for a loooooong time)

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

This is a development issue

### Describe the issue

The game sets a wide path flag for path tiles when a path is wider than 1 tile, to allow easier pathfinding for guests. In vanilla RCT2, this was done in a grid pattern. In OpenRCT2, more tiles get a wide path flag which decreases the amount of junctions. However, it does seem to lead to some situations where the guests will not be able to pathfind past a junction at all. I have encountered several situations like these when playing scenarios which leads to lost guests, which in turn kills your park rating. Here are some examples (white is impassable tiles):

![pathfinding1](https://github.com/user-attachments/assets/da145ff6-7fc2-4f71-b05d-6122773289b7)
![pathfinding2](https://github.com/user-attachments/assets/0793639d-1640-448e-ae78-f982db9189f7)
![pathfinding3](https://github.com/user-attachments/assets/a983ef87-5efc-4028-bf0d-121a21abadfe)
![pathfinding4](https://github.com/user-attachments/assets/d35eb028-871e-4393-a038-167af13e8d53)

Next one is not impassable, but it's a situation where guests are forced to take a looooong way around to get to their destination:

![pathfinding5](https://github.com/user-attachments/assets/6401e22f-fa88-4878-8910-c129dbfe85ef)

The algorithm to set wide path flags may require a look-over.

### Steps to reproduce

Build paths in one of the above configurations. May depend on the orientation of the map.

**Comment by deurklink** [comment]
void FootpathUpdatePathWideFlags(const CoordsXY& footpathPos) in Footpath.cpp is where the wide path flags get set.

**Comment by AaronVanGeffen** [comment]
> This was introduced between v0.0.4 and v0.0.5. In 0.0.4 and prior (incl vanilla) the guests can find the exit without any issues.

Wow. It's super interesting to see what exactly changed between those two versions, and how the code literally doesn't appear to have been touched since.

I'll do some more digging, trying to trace this back to a PR or commit.

**Comment by AaronVanGeffen** [comment]
Some notes / code archaeology:

1. This behaviour appears to be the result of #4527, specifically bdb8a1590361926ff4048f86f68f13c885291217. Wonder if someone on Windows could compile this commit and its parent to confirm?
2. The function was later refactored by #8618 for constants, variable naming. This also undid some of the optimisations by replacing `(true || expr)` with just `expr`, but left their comments.
3. Some initialisation within the function was refactored by #10574 to reduce code duplication, which also changed the element indexes.

**Comment by AaronVanGeffen** [comment]
Turns out I wrongly assumed the issue to be due to optimisations from #4527. I've bisected the issue to 86be86331e53eb7c3a29b6d1d58cc47ab2ada27a, which is from #4221. Bisect log:

```
git bisect start
# status: waiting for both good and bad commits
# bad: [b29be9bfebe36f090b76b2553fc495a72c082e8a] Merge pull request #4527 from zaxcav/fixPathfindHistory
git bisect bad b29be9bfebe36f090b76b2553fc495a72c082e8a
# status: waiting for good commit(s), bad commit known
# bad: [5a773553774fa4d346b5b892bdd4ab03693c88fd] Merge pull request #4548 from janisozaur/compilation-fixes
git bisect bad 5a773553774fa4d346b5b892bdd4ab03693c88fd
# status: waiting for good commit(s), bad commit known
# good: [1eb91cb21044cdd46ac957c956d2ae64255a6bfb] OpenRCT2 0.0.4 (stable)
git bisect good 1eb91cb21044cdd46ac957c956d2ae64255a6bfb
# good: [5c9110b99a8e2760469106052077df352b2d315a] Prevent viewport_invalidate from causing covered viewports to redraw and cache window and viewport visibility.
git bisect good 5c9110b99a8e2760469106052077df352b2d315a
# bad: [ff9e249a8549aaa43aaae853a1db66bf2341e08f] Fix inverse condition of BUILD_SERVER
git bisect good 99a41ee828208ec816a543c216f6c40e422cfa46
# good: [99a41ee828208ec816a543c216f6c40e422cfa46] Fix implicit boolean in GLSL code
git bisect good 99a41ee828208ec816a543c216f6c40e422cfa46
# good: [389d75534d3197532b59845c251bf2285c81cce3] Merge pull request #4224 from marijnvdwerf/clean/data-5
git bisect good 389d75534d3197532b59845c251bf2285c81cce3
# good: [560957824a22966b7a7e2318807112bd60859fe2] Fix #4295: Game crash if the syntax of users.json file is wong
git bisect good 560957824a22966b7a7e2318807112bd60859fe2
# good: [9b77bb35aeafe69e7032bcbf45c25ebcf39f98d9] Fix #4113: Free rides: Income and profit calculated using old price
git bisect good 9b77bb35aeafe69e7032bcbf45c25ebcf39f98d9
# bad: [25328bccc9a14fe7353651f864fd1cf20edb7770] Merge pull request #4221 from zaxcav/pathfind
git bisect bad 25328bccc9a14fe7353651f864fd1cf20edb7770
# bad: [b1da59d25a04de587c059892ab96b3ec6e40043f] Ignore isolated footpath tiles in the wide processing.
git bisect bad b1da59d25a04de587c059892ab96b3ec6e40043f
# bad: [dd32ed3827755b26d554da877229f5ad29a3d127] Revert changes to maxNumJunctions in commit 9c5df6c
git bisect bad dd32ed3827755b26d554da877229f5ad29a3d127
# good: [9bf57074946536d27718782e0e0afbcca0ffb8c9] Rebalance maxNumJunctions to avoid exceeding maxTileChecked limit
git bisect good 9bf57074946536d27718782e0e0afbcca0ffb8c9
# bad: [76974d2d45c9069e57753614bd149d3b7a862078] Rebalance maxNumJunctions - increase value.
git bisect bad 76974d2d45c9069e57753614bd149d3b7a862078
# bad: [86be86331e53eb7c3a29b6d1d58cc47ab2ada27a] Remove exlusions from path tiles being flagged wide based on tile coordinates.
git bisect bad 86be86331e53eb7c3a29b6d1d58cc47ab2ada27a
# first bad commit: [86be86331e53eb7c3a29b6d1d58cc47ab2ada27a] Remove exlusions from path tiles being flagged wide based on tile coordinates.
```

**Comment by AaronVanGeffen** [comment]
Preliminary fix filed in #22827. Please continue discussion there.

**Comment by AaronVanGeffen** [comment]
For changelog purposes, I'm going to consider this a duplicate of #7672.
